libraries/AP_Motors/AP_MotorsMulticopter.cpp

```diff
@@ -409,7 +415,8 @@ void AP_MotorsMulticopter::update_lift_max_from_batt_voltage()
 {
     // sanity check battery_voltage_min is not too small
     // if disabled or misconfigured exit immediately
-    float _batt_voltage_resting_estimate = AP::battery().voltage_resting_estimate(_batt_idx);
+    float _batt_voltage_resting_estimate = has_option(MotorOptions::BATT_RAW_VOLTAGE) ? AP::battery().voltage(_batt_idx) : AP::battery().voltage_resting_estimate(_batt_idx);
+
     if ((_batt_voltage_max <= 0) || (_batt_voltage_min >= _batt_voltage_max) || (_batt_voltage_resting_estimate < 0.25f * _batt_voltage_min)) {
```

Comment:
Is there a case where the measured voltage would now go under 0.25*min? In that case I think it would be better to contain to min rather than 1. `_batt_voltage_filt.reset(_batt_voltage_min / _batt_voltage_max);`

Reply:
This was done deliberately. The only reason for the battery voltage to go below 25% is if the battery monitor has been disconnected. In that case the safest option is to assume the battery is at maximum voltage.

Reply:
I was just thinking with raw voltage we could get extreme sag, if we were to hit this case it would be a discontinuity in outputs. As you say it is quite unlikely, certainly would be a unhappy battery. Probably more likely if your using ESC feedback as the voltage source as that would give the maximum cabling (resistance) between battery and measurement.

Reply:
Sorry, I got distracted.

Maximum gain is: (4.35/2.5)^2 = 3.0276 or more reasonably (4.35/3.3)^2 = 1.7376

So my thoughts are the biggest risk to an aircraft is a failed voltage monitor outputting zero volts that causes the control loops to oscillate.

I wonder if we have a flag for a failed or lost CAN voltage monitor.....